packages/inertia/src/inertia.js

```diff
@@ -170,8 +226,12 @@ export default {
           return this.setPage(error.response.data, { visitId })
         } else if (Axios.isCancel(error)) {
           onCancel()
-        } else if (this.isHardVisit(error.response)) {
-          this.hardVisit(error.response.headers['x-inertia-location'])
+        } else if (this.isLocationVisitResponse(error.response)) {
+          let locationUrl = this.normalizeUrl(error.response.headers['x-inertia-location'])
+          if (url.hash && this.urlWithoutHash(url) === this.urlWithoutHash(locationUrl)) {
+            locationUrl.hash = url.hash
+          }
+          this.locationVisit(locationUrl.href, preserveScroll)
```

Comment:
```suggestion
          const locationUrl = this.normalizeUrl(error.response.headers['x-inertia-location'])
          this.locationVisit(locationUrl.href, preserveScroll)
```

Reply:
The `hash` is included in the `href`. This is needed, as imagine you're making a request to `/users#create`, and a location visit (409) comes back. If we do a `window.location.href` visit there using the `X-Inertia-Location` header, need to add the hash back on, otherwise we'd only visit `/users`. Unless, of course, I'm missing something here.

Reply:
Nevermind, you answered my question already: 

![image](https://user-images.githubusercontent.com/1752195/96255163-bdefb580-0fb6-11eb-8a99-95d608a27421.png)